Description: CVE-2025-32462 - Only allow a remote host to be specified when listing privileges.
 This fixes a bug where a user with sudoers privileges on a different
 host could execute a command on the local host, even if the sudoers
 file would not otherwise allow this. A remote host (-h) may only be
 specified together with the list (-l) or check (-v) operations.
 .
 Backported to sudo 1.8.31 (Ubuntu 20.04): the upstream fix lives in
 sudoers_check_common() using the sudoers_context struct introduced in
 sudo 1.9.14. In 1.8.31 the equivalent logic resides in
 sudoers_policy_main() and uses the global sudo_mode, user_runhost and
 user_host state, so the guard is adapted accordingly. SLOG_AUDIT does
 not exist in this release, so the failure is recorded via
 audit_failure() and log_warningx() as done elsewhere in this function.
Origin: upstream, commit:d530367828e3713d09489872743eb92d31fb11ff
Author: Todd C. Miller <Todd.Miller@sudo.ws>
Last-Update: 2026-07-29
---
--- a/plugins/sudoers/sudoers.c
+++ b/plugins/sudoers/sudoers.c
@@ -298,6 +298,19 @@ sudoers_policy_main(int argc, char * const argv[], int pwflag, char *env_add[],
 	}
     }
 
+    /* The user may only specify a host for "sudo -l" or "sudo -v". */
+    if (!ISSET(sudo_mode, MODE_LIST|MODE_CHECK)) {
+	if (strcmp(user_runhost, user_host) != 0) {
+	    audit_failure(NewArgc, NewArgv,
+		N_("user not allowed to set remote host for command"));
+	    log_warningx(SLOG_NO_STDERR|SLOG_SEND_MAIL,
+		N_("user not allowed to set remote host for command"));
+	    sudo_warnx("%s",
+		U_("a remote host may only be specified when listing privileges."));
+	    goto bad;
+	}
+    }
+
     /* If given the -P option, set the "preserve_groups" flag. */
     if (ISSET(sudo_mode, MODE_PRESERVE_GROUPS))
 	def_preserve_groups = true;
